### DISCUSSION TITLE
netbox and netbox-rq services

### DISCUSSION THREAD

**Discussion by gignat88** [netbox-community/netbox#18103] [Help Wanted!] [answered]
Hello,

I have followed official site for installing netbox ( i have installed version 4.0.11 because i have some plugins that work with what version ).

I start the process from developer mode witch is not recomanded in production. I want do have a service that always start with operating system . I followed also step 4a. and 4b. but although the process started, i cannot open the webpage.

Can anyone provide me a full tutorial of how to create a service that works ?

Thank you in advance.

**Comment by Zombie-Toad** [comment]
Hey,
you should only run 4a OR 4b. After you did one of them you also have to follow step 5 (install and configure a webserver, either apache2 or nginx).
I followed step 4a (Gunicorn) and 5 with nginx and the server runs perfectly.

**Comment by gignat88** [comment]
Hello,

Thank you for answering.

I have followed steps 4a and 5 with nginx and all work fine until the last step with the nginx service restart where i've received an error :

● netbox.service - NetBox WSGI Service
     Loaded: loaded (/etc/systemd/system/netbox.service; enabled; preset: enabled)
     Active: active (running) since Sun 2024-12-01 20:33:30 EET; 5min ago
       Docs: https://docs.netbox.dev/
   Main PID: 1768 (gunicorn)
      Tasks: 6 (limit: 8700)
     Memory: 559.0M (peak: 559.7M)
        CPU: 14.708s
     CGroup: /system.slice/netbox.service
             ├─1768 /opt/netbox/venv/bin/python3 /opt/netbox/venv/bin/gunicorn --pid /var/tmp/netbox.pid --pythonpath /opt/n>
             ├─1781 /opt/netbox/venv/bin/python3 /opt/netbox/venv/bin/gunicorn --pid /var/tmp/netbox.pid --pythonpath /opt/n>
             ├─1782 /opt/netbox/venv/bin/python3 /opt/netbox/venv/bin/gunicorn --pid /var/tmp/netbox.pid --pythonpath /opt/n>
             ├─1783 /opt/netbox/venv/bin/python3 /opt/netbox/venv/bin/gunicorn --pid /var/tmp/netbox.pid --pythonpath /opt/n>
             ├─1784 /opt/netbox/venv/bin/python3 /opt/netbox/venv/bin/gunicorn --pid /var/tmp/netbox.pid --pythonpath /opt/n>
             └─1789 /opt/netbox/venv/bin/python3 /opt/netbox/venv/bin/gunicorn --pid /var/tmp/netbox.pid --pythonpath /opt/n>

the service is UP, but nginx has error  :  

 nginx.service - A high performance web server and a reverse proxy server
     Loaded: loaded (/usr/lib/systemd/system/nginx.service; enabled; preset: enabled)
     Active: failed (Result: exit-code) since Sun 2024-12-01 20:33:30 EET; 6min ago
       Docs: man:nginx(8)
    Process: 1769 ExecStartPre=/usr/sbin/nginx -t -q -g daemon on; master_process on; (code=exited, status=1/FAILURE)
        CPU: 19ms

Dec 01 20:33:30 pici-J42-Nano systemd[1]: Starting nginx.service - A high performance web server and a reverse proxy server.>
Dec 01 20:33:30 pici-J42-Nano nginx[1769]: 2024/12/01 20:33:30 [emerg] 1769#1769: cannot load certificate "/etc/ssl/certs/ne>
Dec 01 20:33:30 pici-J42-Nano nginx[1769]: nginx: configuration file /etc/nginx/nginx.conf test failed
Dec 01 20:33:30 pici-J42-Nano systemd[1]: nginx.service: Control process exited, code=exited, status=1/FAILURE
Dec 01 20:33:30 pici-J42-Nano systemd[1]: nginx.service: Failed with result 'exit-code'.
Dec 01 20:33:30 pici-J42-Nano systemd[1]: Failed to start nginx.service - A high performance web server and a reverse proxy >


2024/12/01 20:41:01 [emerg] 5059#5059: cannot load certificate "/etc/ssl/certs/netbox.crt": BIO_new_file() failed (SSL: error:80000002:system library::No such file or directory:calling fopen(/etc/ssl/certs/netbox.crt, r) error:10000080:BIO routines::no such file)
nginx: configuration file /etc/nginx/nginx.conf test failed

sudo systemctl restart nginx
Job for nginx.service failed because the control process exited with error code.
See "systemctl status nginx.service" and "journalctl -xeu nginx.service" for details.


i have no certificate, how can i solve this ?

thx

**Reply by Zombie-Toad** [reply, marked as answer]
It's right on top of point 5 (Obtain an SSL Certificate):
https://netboxlabs.com/docs/netbox/en/stable/installation/5-http-server/

This is the command:
```bash
sudo openssl req -x509 -nodes -days 365 -newkey rsa:2048 \
-keyout /etc/ssl/private/netbox.key \
-out /etc/ssl/certs/netbox.crt
```

**Reply by gignat88** [reply]
Yes, now it works. Thx a lot for help. You guys are great.

**Reply by gignat88** [reply]
quick question : although i have created a certificate with that command, why it still shows that the connection is not secure ?

![image](https://github.com/user-attachments/assets/d6363a93-d89e-4017-b09d-94911a018ece)

![image](https://github.com/user-attachments/assets/e9c54c4a-afec-401f-bdbe-f95fe98cb74f)

and after this certificate expires i need to enter the same command for creating a new certificate ?

**Reply by candlerb** [reply]
> although i have created a certificate with that command, why it still shows that the connection is not secure ?

Because your certificate is self-signed. Since no third-party has vouched for the identity of your server, the browser has no way to know whether the host you're connecting to is really the correct one.

To fix it you'll need to access your server by domain name (not IP address), and you need get a valid certificate signed by a public CA. You can get these for free from Letsencrypt or ZeroSSL or others, using tools such as certbot or acme.sh or [others](https://letsencrypt.org/docs/client-options/). You need to prove ownership of the domain, either by allowing incoming connections on port 80 from the Internet, or by inserting a TXT record into the DNS zone on request. Also, you'll need to renew it every 90 days (hence set up an automated job for renewal).

All this is nothing specific to Netbox.
